vaadin-spring/src/main/java/com/vaadin/flow/spring/security/RequestUtil.java

```diff
@@ -0,0 +1,41 @@
+package com.vaadin.flow.spring.security;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.vaadin.flow.server.HandlerHelper;
+import com.vaadin.flow.spring.VaadinConfigurationProperties;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Contains utility methods related to request handling.
+ */
+@Component
+public class RequestUtil {
+
+    @Autowired
+    private VaadinConfigurationProperties configurationProperties;
+
+    /**
+     * Checks whether the request is an internal request.
+     *
+     * An internal request is one that is needed for all Vaadin applications to
+     * function, e.g. UIDL or init requests.
+     *
+     * Note that bootstrap requests for any route or static resource requests are
+     * not internal, neither are resource requests for the JS bundle.
+     *
+     * @param request the servlet request
+     * @return {@code true} if the request is Vaadin internal, {@code false}
+     *         otherwise
+     */
+    public boolean isFrameworkInternalRequest(HttpServletRequest request) {
```

Comment:
Could have a unit test for this public method?

Reply:
Or do we actually need this RequestUtil? the JavaDoc in HandlerHelper says
```
@param servletMappingPath
     *            the path the Vaadin servlet is mapped to, with or without and
     *            ending "/*"
```
If the JavaDoc in https://github.com/vaadin/flow/pull/10363 is true, then we can use HandlerHelper directly without of RequestUtil.

Reply:
You need to pass the servlet mapping that is used in Spring to ensure that a request to another servlet is not interpreted as a Vaadin internal request. I will add a unit test

Reply:
OK, but if statement is not necessary?
```
if (vaadinMapping.endsWith("/*")) {
    vaadinMapping = vaadinMapping.substring(0, vaadinMapping.length() - "/*".length());
}
```
based on what I read from the code, then "/*" ending is handled already in HandlerHelper?

Reply:
Removed

Reply:
and added unit test